### PR TITLE
Cleaned up content/unnecessary logging - Emissary Logs

### PR DESCRIPTION
**Pull request by sambish5**
Cleaned up content/unnecessary logging found throughout the entirety of the Emissary program. Any classes that were altered, I went through and parameterized all logs in that given class.

_*Any files with open content logging PRs not included*_

### CODE REVIEW

File: src/main/java/emissary/kff/KffChainLoader.java [hunk 1]
```diff
@@ -53,7 +53,6 @@ public static synchronized KffChain getChainInstance() {
                 logger.debug("No configuration for Known File Filter. Continuing...");
             }
             theInstance = chain;
-            logger.debug("KFF Chain loaded with " + theInstance.size() + " filter using algorithms " + theInstance.getAlgorithms());
```
Comment (jpdahlke) [inline src/main/java/emissary/kff/KffChainLoader.java:56]: This one seems ok to me just parameterize?

File: src/main/java/emissary/kff/KffChainLoader.java [hunk 2]
```diff
@@ -79,24 +78,22 @@ private static int loadFrom(KffChain chain, Map<String, String> m, int kffType,
...
+                // see if known KffType
                 if (kffType == FILE_TYPE) {
                     try {
                         k = (KffFilter) Factory.create(clazz, new Object[] {name, key, filterType});
                     } catch (Exception x) {
                         logger.warn("Cannot create KffFilter, using default", x);
                         k = new KffFile(name, key, filterType);
                     }
-                } else {
-                    logger.error("Unknown kff type " + kffType);
```
Comment (jpdahlke) [inline src/main/java/emissary/kff/KffChainLoader.java:94]: This one also seems ok, just needs an update.

File: src/main/java/emissary/output/filter/AbstractFilter.java [hunk 3]
```diff
@@ -23,9 +23,6 @@
  * Provides the base mechanism for a drop off filter
  */
 public abstract class AbstractFilter implements IDropOffFilter {
-    /** A static convenience logger */
-    protected static Logger slogger = LoggerFactory.getLogger(AbstractFilter.class);
```
Comment (jpdahlke) [inline src/main/java/emissary/output/filter/AbstractFilter.java:27]: These don't seem to have content, just types, etc. I might have to revisit what we say is content, but I'm not sure this class counts.

File: src/main/java/emissary/parser/FillingNIOParser.java [hunk 4]
```diff
@@ -45,15 +39,14 @@ protected byte[] nextChunkOrDie(byte[] data) throws ParserException {
             b.limit(writeOffset);
             b.compact(); // does not re-allocate the byte array, only manipulates the current buffer.
 
+            // buffer has been compacted
             chunkStart += sessionStart;
             sessionStart = 0;
             writeOffset = b.position();
-            logger.debug("Compacted buffer: sessionStart/chunkStart/writeOffset = {}/{}/{}", sessionStart, chunkStart, writeOffset);
```
Comment (jpdahlke) [inline src/main/java/emissary/parser/FillingNIOParser.java:51]: Aren't these positional only?

**Review by jpdahlke — CHANGES_REQUESTED** [review]
I'd make another pass over these files as I think we are being to aggressive in the removals. I added some comments to that effect.

Does anyone else care to weigh in on their expectations to help guide this effort?

File: src/main/java/emissary/command/validator/ServerModeValidator.java [hunk 5]
```diff
@@ -16,7 +16,7 @@ public void validate(String name, String value) throws ParameterException {
             case "standalone":
                 break;
             default:
-                LOG.error("Unknown mode: {}", value);
+                // LOG.error("Unknown mode: {}", value);
```
Comment (dev-mlb) [inline src/main/java/emissary/command/validator/ServerModeValidator.java:19]: you can remove this and the logger declaration on line 10

**Review by dev-mlb — DISMISSED** [review]

**Comment by jpdahlke** [comment]
Circling back to this PR, I think I'm good with everything except:

- revert ServerModeValidator.java back to original form (this message is helpful)
- ServiceConfigGuide should only be printing key/value information provided in the config files themselves so I think it's safe to leave the log contents alone
- In HDMobileAgent, the `logger.isDebugEnabled()` blocks can be restored. They are helpful for routing and should only contain proxy information, not content.
- TikaFilePlace/UnixFilePlace, restore the commented out lines and their content. Identification information like this should be ok.
- Restore removed statements in AbstractFilter
- Restore the large `logger.isDebugEnabled()` block in ConstructorLookupCache for types/params
- Restore the removed statements from the TypeEngine class. These are helpful when wiring engines up.

For anything you restore from the above list, the reformatting/parameterization changes of the statements are fine.

File: src/main/java/emissary/config/ServiceConfigGuide.java [hunk 6]
```diff
@@ -310,7 +310,7 @@ protected ConfigEntry handleNewEntry(final String parmNameArg, final String sval
             if (fileFlavors != null && fileFlavors.length > 0) {
                 fileFlavorList.addAll(Arrays.asList(fileFlavors));
             }
-            logger.debug("ServiceConfigGuide::handleNewEntry -- FileFlavorList = " + fileFlavorList);
+            logger.debug("ServiceConfigGuide::handleNewEntry -- FileFlavorList = _+{}", fileFlavorList);
```
Comment (jpdahlke) [inline src/main/java/emissary/config/ServiceConfigGuide.java:313]: This looks like a typo?

**Review by drivenflywheel — APPROVED** [review]
Looks good (aside from the one remaining typo that Jason noted)

**Review by jpdahlke — APPROVED** [review]

**Pull request merged**